runtime/src/certify_key_extended.rs

```diff
@@ -0,0 +1,121 @@
+/*++
+
+Licensed under the Apache-2.0 license.
+
+File Name:
+
+    populate_idev.rs
+
+Abstract:
+
+    File contains CertifyKeyExtended mailbox command.
+
+--*/
+
+use core::str::from_utf8;
+
+use arrayvec::ArrayVec;
+use bitflags::bitflags;
+use caliptra_common::mailbox_api::{
+    CertifyKeyExtendedReq, CertifyKeyExtendedResp, MailboxResp, MailboxRespHeader,
+};
+use caliptra_error::{CaliptraError, CaliptraResult};
+use dpe::{
+    commands::{CertifyKeyCmd, Command, CommandExecution},
+    response::Response,
+};
+use zerocopy::{AsBytes, FromBytes};
+
+use crate::{
+    CptraDpeTypes, DpeCrypto, DpeEnv, DpePlatform, Drivers, MAX_CERT_CHAIN_SIZE, PL0_PAUSER_FLAG,
+};
+
+#[repr(C)]
+#[derive(Debug, PartialEq, Eq, FromBytes, AsBytes)]
+pub struct CertifyKeyExtendedFlags(u32);
+
+bitflags! {
+    impl CertifyKeyExtendedFlags: u32 {
+        const DMTF_OTHER_NAME = 1u32 << 31;
+    }
+}
+
+pub struct CertifyKeyExtendedCmd;
+impl CertifyKeyExtendedCmd {
+    pub(crate) fn execute(drivers: &mut Drivers, cmd_args: &[u8]) -> CaliptraResult<MailboxResp> {
+        if let Some(cmd) = CertifyKeyExtendedReq::read_from(cmd_args) {
+            let hashed_rt_pub_key = drivers.compute_rt_alias_sn()?;
+            let key_id_rt_cdi = Drivers::get_key_id_rt_cdi(drivers)?;
+            let key_id_rt_priv_key = Drivers::get_key_id_rt_priv_key(drivers)?;
+            let pdata = drivers.persistent_data.get_mut();
+            let crypto = DpeCrypto::new(
```

Comment:
nit: Can we share more of the code for initializing DpeCrypto and DpePlatform? Maybe with a helper in drivers.

Reply:
I tried doing this earlier but ran into some issues with the borrow checker. Will try again in a followup PR since this one is already big.